### ISSUE TITLE
📎 Implement linter rule `noNamespace`, `no-namespace`

### ISSUE THREAD

**Issue by Conaclos** [rome/tools#4274]
### Description

Implement ESLint rule [no-namespace](https://typescript-eslint.io/rules/no-namespace/).

**Comment by noftaly** [comment]
Hey there! I'd like to try this rule if it's possible 😊 

I've noticed that the `typescript-eslint` rule you linked does a few things:
- it is disable-able for declarations (`declare namespace Foo {}`) [enabled by default]
- it is disabled by default for definition files (`.d.ts`)
- it does not report for external modules `declare module 'foo' {}`

Which of those behaviours would you like to replicate exactly? aka, how to treat namespace declarations, namespaces in definition files, and external module redeclaration?

**Comment by Conaclos** [comment]
Hi @noftaly!

> it is disable-able for declarations (declare namespace Foo {}) [enabled by default]

I think the rule should enable this syntax by default.

> it does not report for external modules declare module 'foo' {}

In the same vein, this should be enabled.

> it is disabled by default for definition files (.d.ts)

Unfortunately, _Rome_ is for now not able to query the file type.
Checking to the output of TypeScript, it seems that TypeScript outputs `declare namespace` for a `namesapce` in its declaration file.

It is still possible that some dev hand-crafted their declaration files with a `,amespace` without `declare`.
We should enable this later when _Rome_ will support to query file extension in a linter rule.

Let's me know if you are still interested :)

**Comment by noftaly** [comment]
> I think the rule should enable this syntax by default.
> In the same vein, this should be enabled.

I understand, so the rule will report for all those cases!

> Unfortunately, Rome is for now not able to query the file type.

Ok, we'll consider that when Rome supports file types 😄 

> Let's me know if you are still interested :)

I am!
I just started playing around, and I've made the simplest check possible, just looking for `TsModuleDeclaration`. It looks like it doesn't trigger for `declare module 'foo' {}`, although, when checking on ASTExplorer with `@typescript-eslint`'s parser, it is in fact a `TsModuleDeclaration`. Is this a problem on my side? or is it named differently in Rome?
